### PR TITLE
New ColdFusion 2018 functions

### LINKED ISSUE #887: New ColdFusion 2018 functions
The following functions are new to ColdFusion 2018. Several were already supported by Lucee, so the engine information just needs to be updated.

- [arrayFirst](https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-a-b/arrayfirst.html) (this already exists for Lucee)
- [arrayLast](https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-a-b/arraylast.html) (this already exists for Lucee)
- cacheGetEngineProperties (no doc found): Returns information about the [cache engine](https://helpx.adobe.com/coldfusion/developing-applications/changes-in-coldfusion/caching-enhancements.html)
- [queryDeleteColumn](https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-m-r/querydeletecolumn.html) (this already exists for Lucee)
- [queryDeleteRow](https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-m-r/querydeleterow.html) (this already exists for Lucee)
- [runAsync](https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-m-r/runasync.html)
- [structIsOrdered](https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-s/structisordered.html)

### PR DESCRIPTION
**Pull request by shaedrich**
Resolve #887

### CODE REVIEW

File: data/en/runasync.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+    "name":"runAsync",
+    "type":"function",
+    "syntax":"runAsync(function() {})",
+    "script":"runAsync(function() {});",
+    "returns":"Future",
```
Comment (KamasamaK) [inline data/en/runasync.json]: I do not see that they added this as a new return type. This should probably be `any`.

Comment (shaedrich) [reply data/en/runasync.json]: They say, it is called `Future` which is something like a `Promise` in JavaScript. Should we really use something unspecific like `any`?

Comment (ddspringle) [reply data/en/runasync.json]: @shaedrich To be clear, `Future` in CFML is nothing like a `Promise` in JavaScript (its a subtle difference, but is different enough not to confuse the two). It is closer to a `CompletableFuture` in Java. 

That said, and in this case, I think `any` is probably appropriate here given it returns any value... it just happens to return it at a future point in the code (which could perhaps be spelled out in the description, if it isn't already)

**Comment by pfreitag** [comment]
Nice work - thanks!

**Comment by shaedrich** [comment]
My pleasure!

**Comment by KamasamaK** [comment]
I still don't think it makes sense to use `script` for anything but tags.

**Comment by pfreitag** [comment]
@KamasamaK - not sure what you were referring to here?

**Comment by pfreitag** [comment]
Ah yes - I agree - that is only necessary for tags to show the cfscript equivalent.

**Pull request merged**